### PR TITLE
Update to nodejs 16 and npm lockfile version 2

### LINKED ISSUE #2450: Update to node 16 / npm 7
**Is your feature request related to a problem? Please describe.**
Our current tooling for the nodejs sdk and example game server use node 12 and npm 6. 

According to https://nodejs.org/en/about/releases/ node 12 is in maintenance mode only until summer 2022 and we should update to the active version.

**Describe the solution you'd like**
Updating the versions should make it easier to keep the version dependencies in the package-lock.json files for the example game server and sdk in sync.

**Describe alternatives you've considered**
Leave it as-is until we are forced to upgrade due to support expiring. 

**Additional context**
https://github.com/googleforgames/agones/pull/2447

@steven-supersolid

### PR DESCRIPTION
**Pull request by steven-supersolid**
**What type of PR is this?**
/kind cleanup

**What this PR does / Why we need it**:
* Update to Node.js 16.x as this is the current LTS version
* Update also includes npm update to 8.x (8.1.2 bundled with node 16.13.2 used to perform the package-lock upgrade)
* Update dependencies
* Resync package-lock file with examples - should be easier to keep in sync now
* Update promisified version of setTimeout in example to use that included in Node.js 16

**Which issue(s) this PR fixes**:

Closes #2450

**Special notes for your reviewer**:

### CODE REVIEW

File: examples/nodejs-simple/src/index.js [hunk 1]
```diff
@@ -13,9 +13,7 @@
 // limitations under the License.
 
 const AgonesSDK = require('@google-cloud/agones-sdk');
-const util = require('util');
-
-const sleep = util.promisify(setTimeout);
+const {setTimeout} = require('timers/promises');
```
Comment (steven-supersolid) [inline examples/nodejs-simple/src/index.js:16]: This will require using Node.js 16, we could revert if we want the examples to remain backwards compatible

**Comment by steven-supersolid** [comment]
The build failed on this step
```
Step #10 - "lint": Step 34/42 : RUN apt-get update && apt-get install -y nodejs
Step #10 - "lint":  ---> Running in 469ca59598d7
Step #10 - "lint": Hit:1 http://deb.debian.org/debian buster InRelease
Step #10 - "lint": Hit:2 http://deb.debian.org/debian buster-updates InRelease
Step #10 - "lint": Hit:3 http://security.debian.org/debian-security buster/updates InRelease
Step #10 - "lint": Ign:4 https://deb.nodesource.com/node_16.x jessie InRelease
Step #10 - "lint": Err:5 https://deb.nodesource.com/node_16.x jessie Release
Step #10 - "lint":   404  Not Found [IP: 23.62.230.72 443]
Step #10 - "lint": Reading package lists...
Step #10 - "lint": [91mE: The repository 'https://deb.nodesource.com/node_16.x jessie Release' does not have a Release file.
Step #10 - "lint": The command '/bin/sh -c apt-get update && apt-get install -y nodejs' returned a non-zero code: 100
Step #10 - "lint": make[2]: *** [includes/build-image.mk:26: build-build-image] Error 100
Step #10 - "lint": [0mmake[2]: Leaving directory '/workspace/build'
Step #10 - "lint": make[1]: *** [includes/build-image.mk:54: ensure-image] Error 2
Step #10 - "lint": make[1]: Leaving directory '/workspace/build'
Step #10 - "lint": make: *** [includes/build-image.mk:41: ensure-build-image] Error 2
```

Looking at https://deb.nodesource.com/node_12.x/dists/, there is a `jessie` folder but https://deb.nodesource.com/node_16.x/dists/ doesn't have one although the next version `stretch` is present

**Comment by roberthbailey** [comment]
Debian 8 "jessie" is super old ([initially released in 2015](https://www.debian.org/releases/jessie/)) so I'm not sure why we would still be using it. Our build image is using Debian 10 "buster" (see [here](https://github.com/googleforgames/agones/blob/main/build/build-image/Dockerfile#L17)). 

Any idea why it switches from buster

```
Step #10 - "lint": Hit:3 http://security.debian.org/debian-security buster/updates InRelease
```

to jessie

```
Step #10 - "lint": Ign:4 https://deb.nodesource.com/node_16.x jessie InRelease
```

in the following step?

**Comment by steven-supersolid** [comment]
It looks like the version arg was introduced in #2438 and set differently

**Comment by steven-supersolid** [comment]
From https://github.com/nodesource/distributions/blob/master/README.md#manual-installation it looks like we could alternatively we could replace the manual arg setting with the following if that will execute correctly in the Dockerfile?
```
ARG DISTRO="$(lsb_release -s -c)"
```

**Comment by roberthbailey** [comment]
@cindy52 might have some context on the change in #2438

**Comment by steven-supersolid** [comment]
Tests failing due to an expired certificate on a 3rd party website:
```
for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20; \
	do echo "Html Test: Attempt $i" && \
	  docker run --rm -t -e "TERM=xterm-256color" -v /workspace/build//.config/gcloud:/root/.config/gcloud -v ~/.kube/:/root/.kube -v ~/.config/helm:/root/.config/helm -v ~/.cache/helm:/root/.cache/helm -v /workspace:/go/src/agones.dev/agones -v /workspace/build//.gomod:/go/pkg/mod -v /workspace/build//.gocache:/root/.cache/go-build  agones-build:02a242a0ad bash -c \
		"mkdir -p /tmp/website && cp -r /go/src/agones.dev/agones/site/public /tmp/website/site && htmltest -c /go/src/agones.dev/agones/site/htmltest.yaml /tmp/website" && \
break || sleep 60 && false; done
Html Test: Attempt 1
htmltest started at 08:02:16 on /tmp/website
========================================================================
site/index.html
  Get https://www.embark.games/: x509: certificate has expired or is not yet valid --- site/index.html --> https://www.embark.games/
========================================================================
✘✘✘ failed in 12m54.620830518s
```

**Comment by roberthbailey** [comment]
You can add a `data-proofer-ignore` to it like we have for some of the other external links on that page.

**Comment by steven-supersolid** [comment]
The certificate looks to be renewed so will try another test

File: examples/nodejs-simple/src/index.js (hunk 1 above)
Comment (roberthbailey) [reply examples/nodejs-simple/src/index.js:16]: I'd be more concerned if the SDK code required the new version but I think it's ok for the example code to use the new version.

**Review by roberthbailey — APPROVED** [review]

**Comment by cindy52** [comment]
> From https://github.com/nodesource/distributions/blob/master/README.md#manual-installation it looks like we could alternatively we could replace the manual arg setting with the following if that will execute correctly in the Dockerfile?
> 
> ```
> ARG DISTRO="$(lsb_release -s -c)"
> ```

The arg doesn't work as I tried before and I need to provide a specific DISTRO.

**Comment by cindy52** [comment]
> Debian 8 "jessie" is super old ([initially released in 2015](https://www.debian.org/releases/jessie/)) so I'm not sure why we would still be using it. Our build image is using Debian 10 "buster" (see [here](https://github.com/googleforgames/agones/blob/main/build/build-image/Dockerfile#L17)).
> 
> Any idea why it switches from buster
> 
> ```
> Step #10 - "lint": Hit:3 http://security.debian.org/debian-security buster/updates InRelease
> ```
> 
> to jessie
> 
> ```
> Step #10 - "lint": Ign:4 https://deb.nodesource.com/node_16.x jessie InRelease
> ```
> 
> in the following step?

We can change the DISTRO ARG from jessie to buster [here](https://github.com/googleforgames/agones/pull/2438/files#diff-686b06cb4b89c224b0df5f76fa2bbc7b5757477074e6917a21b8040a9e6857a4R97).

**Pull request merged**